### PR TITLE
Anchor signer

### PR DESCRIPTION
**Pull request by roeierez**
This PR upgrades the greenlight dependency to the one that supports zero fee htlc: https://github.com/Blockstream/greenlight/pull/237

Since there was a breaking change there that changes the `bip32_ext_key` value we also needed to fallback to the old key in some cases. We will need to stay with this code for some reasonable time.

### CODE REVIEW

**Comment by ok300** [comment]
Starting when (date, commit, etc) did GL change the `bip32_ext_key` implementation?

I'm planning to restore a backup from before that date, using this code (when PR is ready), to test if the fallback works.

File: libs/sdk-core/src/breez_services.rs [hunk 1]
```diff
@@ -1117,7 +1117,7 @@ impl BreezServices {
             .target(env_logger::Target::Pipe(target_log_file))
             .parse_filters(
                 r#"
-                info,
+                debug,
```
Comment (JssDWt) [inline libs/sdk-core/src/breez_services.rs:1120]: don't forget to undo this

**Review by JssDWt — APPROVED** [review]
It looks good. As long as the upgraded client is the only client. Should there be any legacy clients too for the same node, this is a problem, because then we're encrypting the backup with an unknown key to that client.

**Comment by roeierez** [comment]
> It looks good (assuming it's tested). As long as the upgraded client is the only client. Should there be any legacy clients too for the same node, this is a problem, because then we're encrypting the backup with an unknown key to that client.

Right but currently it won't run side by side even because of the signer incompetability.

**Review by dangeross — APPROVED** [review]
LGTM! 

Were you able to test if an anchor channel was created?

**Comment by roeierez** [comment]
> LGTM!
> 
> Were you able to test if an anchor channel was created?

Yes, was created.

**Pull request merged**